### DISCUSSION TITLE
Destroy modal content on hide

### DISCUSSION THREAD

**Discussion by cubecleveland** [ariakit/ariakit#933] [Questions] [answered]
Hello, we use the modal compoenent to show a video element that also streams audio.
We want to actually destory the elment inside the modal when the modal is hidden...
IS there any way to do this automatically, 
We want to still be able to close the modal on clicking outside ( on the backdrop)......

**Comment by diegohaz** [comment, marked as answer]
Yes! `Dialog` uses `DisclosureContent` underneath, so you can follow the approach on https://reakit.io/docs/disclosure/#conditionally-rendering
